community/kernel/src/main/java/org/neo4j/graphdb/traversal/BranchCollisionPolicy.java

```diff
@@ -19,10 +19,37 @@
  */
 package org.neo4j.graphdb.traversal;
 
+import org.neo4j.function.Predicate;
+import org.neo4j.graphdb.Path;
+
 /**
  * Copied from kernel package so that we can hide kernel from the public API.
  */
 public interface BranchCollisionPolicy
 {
+    @Deprecated
     BranchCollisionDetector create( Evaluator evaluator );
+
+    BranchCollisionDetector create( Evaluator evaluator, Predicate<Path> pathPredicate );
+
+    class CollisionPolicyAdapter implements BranchCollisionPolicy
```

Comment:
Indeed that is kind adapter/wrapper thingy which is needed to preserve cypher 1.9 compatibility. We need to wrap collision policy coming from cypher to adapt it to new interface otherwise it will fail with `AbstractMethod` method exception